tests/tests/json_pushdown.rs

```diff
@@ -0,0 +1,594 @@
+                    "name",
+                    "age",
+                    "theme",
+                    "level2",
+                    "text",
+                    "number",
+                    "boolean",
+                    "null_value"
+                ])
+            )
+                .prop_map(|(key1, key2)| JsonPath::Nested(key1.to_string(), key2.to_string())),
+            (
+                proptest::sample::select(vec!["level1"]),
+                proptest::sample::select(vec!["level2"]),
+                proptest::sample::select(vec!["level3"])
+            )
+                .prop_map(|(key1, key2, key3)| JsonPath::DeepNested(
+                    key1.to_string(),
+                    key2.to_string(),
+                    key3.to_string()
+                )),
+        ]
+        .boxed()
+    }
+}
+
+impl JsonPath {
+    fn is_boolean_field(&self) -> bool {
+        match self {
+            JsonPath::Simple(key) => key == "active",
+            JsonPath::Nested(_, key2) => key2 == "boolean",
+            JsonPath::DeepNested(_, _, _) => false,
+        }
+    }
+
+    fn is_numeric_field(&self) -> bool {
+        match self {
+            JsonPath::Simple(key) => key == "count",
+            JsonPath::Nested(_, key2) => key2 == "age" || key2 == "number",
+            JsonPath::DeepNested(_, _, _) => false,
+        }
+    }
+}
+
+impl JsonPath {
+    fn to_sql(&self) -> String {
+        match self {
+            JsonPath::Simple(key) => format!("'{key}'"),
+            JsonPath::Nested(key1, key2) => format!("'{{{key1},{key2}}}'"),
+            JsonPath::DeepNested(key1, key2, key3) => format!("'{{{key1},{key2},{key3}}}'"),
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub enum JsonOperation {
+    Comparison {
+        operator: Operator,
+        value: JsonValueType,
+    },
+    IsNull,
+    IsNotNull,
+    IsTrue,
+    IsFalse,
+    In {
+        values: Vec<JsonValueType>,
+    },
+    NotIn {
+        values: Vec<JsonValueType>,
+    },
+}
+
+impl Arbitrary for JsonOperation {
+    type Parameters = ();
+    type Strategy = BoxedStrategy<Self>;
+
+    fn arbitrary_with(_args: Self::Parameters) -> Self::Strategy {
+        prop_oneof![
+            // For comparison operations, ensure type compatibility
+            (any::<Operator>(), any::<JsonValueType>())
+                .prop_filter(
+                    "operator and value type must be compatible",
+                    |(operator, value)| { value.is_compatible_with_operator(operator) }
+                )
+                .prop_map(|(operator, value)| JsonOperation::Comparison { operator, value }),
+            Just(JsonOperation::IsNull),
+            Just(JsonOperation::IsNotNull),
+            // Only allow IsTrue/IsFalse for boolean types
+            any::<JsonValueType>()
+                .prop_filter("IsTrue/IsFalse only work with boolean types", |value| value
+                    .is_boolean_type())
+                .prop_map(|_| JsonOperation::IsTrue),
```

Comment:
Hm... not 100% sure here, but is this equivalent:
```suggestion
            Just(JsonOperation::IsTrue),
```
?

If you were actually using the `JsonValueType`, it would be a different story.